contracts/Content/UniqueContent.sol

```diff
@@ -0,0 +1,173 @@
+        emit Mint(_msgSender(), _data);
+
+        // increment uniqueIdsCounter for next minting process
+        uniqueIdsCounter++;
+    }
+
+    /** Asset Burning
+    * @dev If the caller is the owner of the token and if the unique asset is not creator locked, it returns the original asset, burns the unique asset, and then deletes the unique asset's token info
+    * @param _uniqueId uint256 ID of token to burn
+    */
+    function burn(uint256 _uniqueId) external override {
+        require(ownerOf(_uniqueId) == _msgSender(), "You must be the owner of this token to burn it");
+        require(!uniqueAssetInfo[_uniqueId].creatorLocked, "Creator has disabled the burning of this token");
+        // transfers original asset back to caller
+        IERC1155Upgradeable(uniqueAssetInfo[_uniqueId].contentAddress).safeTransferFrom(address(this), _msgSender(), uniqueAssetInfo[_uniqueId].tokenId, 1, "");
+        _burn(_uniqueId);
+
+        uniqueAssetInfo[_uniqueId].creator = address(0);
+        uniqueAssetInfo[_uniqueId].contentAddress = address(0);
+        uniqueAssetInfo[_uniqueId].tokenId = 0;
+        uniqueAssetInfo[_uniqueId].version = 0;
+        uniqueAssetInfo[_uniqueId].creatorLocked = false;
+        delete uniqueAssetInfo[_uniqueId].uniqueAssetUri;
+        delete tokenRoyalties[_uniqueId];
+        
+        emit Burn(_msgSender(), _uniqueId);
+    }
+
+    /**
+    * @dev Returns the uri of a specific version of the original asset the unique asset is based on
+    * @param _uniqueId uint256 ID of token to query original asset uri of
+    * @param _version version number of token to query
+    */
+    function originalAssetUri(uint256 _uniqueId, uint256 _version) external view override returns (string memory) {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        uint256 _tokenId = uniqueAssetInfo[_uniqueId].tokenId;
+        return IContent(uniqueAssetInfo[_uniqueId].contentAddress).uri(_tokenId, _version);
+    }
+
+    /**
+    * @dev Returns the unique asset uri of a specific version
+    * @param _uniqueId uint256 ID of token to query
+    * @param _version version number of token to query
+    */
+    function uri(uint256 _uniqueId, uint256 _version) external view override returns (string memory) {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        if (_version > uniqueAssetInfo[_uniqueId].version) {
+            _version = uniqueAssetInfo[_uniqueId].version;
+        } 
+        return uniqueAssetInfo[_uniqueId].uniqueAssetUri[_version];
+    }
+
+    /**
+    * @dev Adds a new version of the unique asset
+    * @param _uniqueId uint256 ID of the token to set its uri
+    * @param _uri string URI to assign
+    */
+    function setUniqueUri(uint256 _uniqueId, string memory _uri) external override {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        require(uniqueAssetInfo[_uniqueId].creator == _msgSender(), "You are not the creator of this token");
```

Comment:
actually, add a check that setUniqueUri() should only be called if the creator still holds the asset. if they've already sold or given away the asset, they shouldn't be able to update the uri (i.e. risk of a rugpull)